src/System.Xml.XDocument/System/Xml/Linq/XDocument.cs

```diff
@@ -479,6 +599,40 @@ public void Save(Stream stream, SaveOptions options)
         }
 
         /// <summary>
+        /// Output this <see cref="XDocument"/> to a <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        /// The <see cref="Stream"/> to output the XML to.  
+        /// </param>
+        /// <param name="options">
+        /// If SaveOptions.DisableFormatting is enabled the output is not indented.
+        /// If SaveOptions.OmitDuplicateNamespaces is enabled duplicate namespace declarations will be removed.
+        /// </param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        public async Task SaveAsync(Stream stream, SaveOptions options, CancellationToken cancellationToken)
+        {
+            XmlWriterSettings ws = GetXmlWriterSettings(options);
+
+            ws.Async = true;
+
+            if (declaration != null && !string.IsNullOrEmpty(declaration.Encoding))
+            {
+                try
+                {
+                    ws.Encoding = Encoding.GetEncoding(declaration.Encoding);
+                }
+                catch (ArgumentException)
```

Comment:
After looking at the documentation I see that what we throw if the encoding cannot be found and you are doing what we already do in Save. While I don't particularly like this approach I also don't see another option so I guess we will have to live with this.